### PR TITLE
Fix parsing in GoModExtractor

### PR DESCRIPTION
**Pull request by jakeryang**
Addresses #101 

Open to suggestions on how to avoid using `trim()` on every line.

I was thinking about checking if the line is `)` or `)\r` (or maybe even `)\n`) if that's even a case. 

Or just trim all of the lines in the first place?

### CODE REVIEW

File: extractor/src/extractors/GoModExtractor.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export default class GoModExtractor implements Extractor {
 
                 case "require":
                     i++;    // requires on subsequent lines
-                    for (; i < lines.length && lines[i] !== ")"; i++) {
+                    for (; i < lines.length && lines[i].trim() !== ")"; i++) {
```
Comment (mjpitz) [inline extractor/src/extractors/GoModExtractor.ts:51]: Is there a place we can do this once and cache the result (after all, we are doing it in two places right here)

**Review by mjpitz — APPROVED** [review]
One desired improvement, but the code looks good as is. Let me know if there's a good place to cache the result of the trim.

**Comment by mjpitz** [comment]
Pro-tip: use `Resolves gh-##` when opening the PR instead of `Addresses`. This will automatically link the associated ticket and automatically close it when this PR is closed.

File: extractor/src/extractors/GoModExtractor.ts (hunk 1 above)
Comment (jakeryang) [reply extractor/src/extractors/GoModExtractor.ts:51]: I was thinking of trimming each line before we run the main loop using something like this:
`const lines = content.split(/\n+/g).map(i => i.trim());`

**Comment by jakeryang** [comment]
Just committed a change that will trim each line before running the main loop.

File: extractor/src/extractors/GoModExtractor.ts [hunk 2]
```diff
@@ -24,18 +24,17 @@ export default class GoModExtractor implements Extractor {
     public async extract(_: string, files: { [p: string]: ExtractorFile }): Promise<DependencyManagementFile> {
         const content = files["go.mod"].raw();
 
-        const lines = content.split(/\n+/g);
+        const lines = content.split(/\n+/g).map(i => i.trim());
```
Comment (mjpitz) [inline extractor/src/extractors/GoModExtractor.ts:27]: perfect!

**Review by mjpitz — APPROVED** [review]

**Comment by mjpitz** [comment]
If you're stuck on package-lock.json, don't worry about it. I've been going back and forth with it and added it as an explicit exclusion. Should just be the difference on GoModExtractor.

**Comment by mjpitz** [comment]
@jakeryang thanks for your pull request!

**Pull request merged**